script.js

```diff
@@ -0,0 +1,228 @@
+    return word.length;
+}
+
+function getUppercaseLetter(word) {
+    const uppercaseLetters =  /[A-Z]/g;
+    const matchingResult = word.match(uppercaseLetters);
+    if (matchingResult === null) {
+        return 0;
+    }
+    return matchingResult.length;
+}
+
+function getNumber(word) {
+    const numbers = /\d/g;
+    const matchingResult = word.match(numbers);
+    if (matchingResult === null) {
+        return 0;
+    }
+    return matchingResult.length;
+}
+
+function isValidPassword(password) {
+    return getLettersAmount(password) >= 8 &&
+        getUppercaseLetter(password) >= 1 &&
+        getNumber(password) >= 1;
+}
+
+console.log(isValidPassword('Password123'));
+
+// 9. In the getPasswordStrength.js file.
+
+// 10. Create the getTotalOrderCost function that returns the total cost of the order and accepts the following arguments:
+// - the base cost of the items
+// - whether the customer is a VIP member - they get a 5% discount
+// - how many loyalty points the customer wants to use - each one of them reduces the cost by a cent (the final cost should not be lower than $0)
+// - is the package going to be shipped internationally - shipping internationally costs $10 (otherwise, it costs $5)
+// Make sure to apply the VIP discount before the loyalty points. The VIP discount does not apply to shipping.
+
+function isVipSubscriptionActive(vipSubscription) {
+    if (vipSubscription === 'active') {
+        return 0.05;
+    }
+    return 0;
+}
+
+function getLoyaltyPoints(pointAmount) {
+    if (pointAmount) {
+        return pointAmount;
+    }
+    return 0;
+}
+
+function isShippedInternationally(shippingType) {
+    if (shippingType === 'international') {
+        return 10;
+    }
+    return 5;
+}
+
+function getTotalOrderCost(baseCost, vipMember, loyaltyPoints, shipping) {
+    const fivePercentDiscount = isVipSubscriptionActive(vipMember) * baseCost;
+    const costReduceThroughLoyaltyPoints = getLoyaltyPoints(loyaltyPoints) / 100;
+    const shippingType = isShippedInternationally(shipping);
+    return baseCost - fivePercentDiscount - costReduceThroughLoyaltyPoints + shippingType;
+}
+
+console.log(getTotalOrderCost(100,'',5, 'international'));
+
+// 11. Create the getTicketPrice function that returns the ticket price and accepts the following arguments:
+// - the base price of the ticket
+// - how many days are there until the show - buying tickets more than a month in advance provides a 10% discount
+// - is the show on the weekend - weekend shows are $15 more expensive
+// The discount applied if the tickets are bought a month in advance should decrease the $15 charged if the show plays on the weekend.
+
+function getDaysUntilTheShow(daysNumber) {
+    if (daysNumber > 30) {
+        return 0.10;
+    }
+    return 0;
+}
+
+function isTheShowOnTheWeekend(weekendShow) {
+    if (weekendShow === "yes") {
+        return 15;
+    }
+    return 0;
+}
+
+function getTicketPrice(basePrice, daysUntilTheShow, weekendShows) {
```

Comment:
`weekendShows` should be a boolean

Reply:
Also, since it should be a boolean, try starting the variable with the word `is`